pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -54,12 +60,44 @@ func updateConditions(msConditions []v1alpha1.Condition, prom monv1.Prometheus,
 				reconciled.LastTransitionTime = metav1.Now()
 			}
 			updatedConditions = append(updatedConditions, reconciled)
+		case v1alpha1.ResourceDiscoveryCondition:
+			resourceDiscovery := updateResourceDiscovery(ms)
+			if !resourceDiscovery.Equal(mc) {
+				resourceDiscovery.LastTransitionTime = metav1.Now()
+			}
+			updatedConditions = append(updatedConditions, resourceDiscovery)
 		}
 	}
 
 	return updatedConditions
 }
 
+// updateResouceDiscovery updates the ResouceDiscoveryCondition based on the
+// ResrouceSelector in the MonitorinStack spec. A ResourceSelector of nil causes
```

Comment:
Nit: typo in `ResrouceSelector`

Reply:
thanks, nice catch. For some reason typing `resource` is the bane of my existence.